### ISSUE TITLE
Warnings about deprecated builtin with clang 15

### ISSUE THREAD

**Issue by mmuetzel** [ERGO-Code/HiGHS#989]
When building HiGHS 1.3.0 with clang 15, I see loads of deprecated-builtin warnings like the following:
```
D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/util/HighsHash.h:45:53: warning: builtin __has_trivial_copy is deprecated; use __is_trivially_copyable instead [-Wdeprecated-builtins]
struct HighsHashable : std::integral_constant<bool, IS_TRIVIALLY_COPYABLE(T)> {
                                                    ^
D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/util/HighsHash.h:39:34: note: expanded from macro 'IS_TRIVIALLY_COPYABLE'
#define IS_TRIVIALLY_COPYABLE(T) __has_trivial_copy(T)
                                 ^
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/mip/HighsDomain.cpp:13:
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/mip/HighsDomain.h:23:
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/mip/HighsMipSolver.h:16:
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/Highs.h:26:
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/presolve/PresolveComponent.h:27:
In file included from D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/presolve/HighsPostsolveStack.h:31:
D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/util/HighsDataStack.h:40:37: warning: builtin __has_trivial_copy is deprecated; use __is_trivially_copyable instead [-Wdeprecated-builtins]
            typename std::enable_if<IS_TRIVIALLY_COPYABLE(T), int>::type = 0>
                                    ^
D:/MINGW-packages/mingw-w64-highs/src/HiGHS-1.3.0/src/util/HighsDataStack.h:27:34: note: expanded from macro 'IS_TRIVIALLY_COPYABLE'
#define IS_TRIVIALLY_COPYABLE(T) __has_trivial_copy(T)
                                 ^
```

IIUC, there are already some pre-processor conditions that select which builtin to use depending on compiler versions. It looks like that doesn't cover Clang.
Maybe, it would be better to use feature tests instead of compiler version checks?

**Comment by FlorianSchwendinger** [comment]
I see the same message. As far as I see, as the warning suggests replacing
`__has_trivial_copy` with `__is_trivially_copyable` does the trick on
macos-highsierra --- AppleClang 10.0.0.10001145
Debian --- Clang 14.0.6
Debian --- GNU 10.2.1
Windows Server 2022 --- GNU 10.4.0
Fedora --- Clang 11.0.0

**Comment by jajhall** [comment]
Fixed by 271c1dd
